code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -1704,12 +1700,12 @@ pub(crate) fn do_reward_accumulation<T: Config>(
 		let releasable_periods_surpassed =
 			cmp::min(maximum_releasable_periods, periods_surpassed.into());
 
-		// saturating is safe here since these values are checked against max_rewards anyways, which
-		// is <= u128::MAX
-		let newly_accumulated_rewards =
-			u128::saturating_mul(releasable_periods_surpassed, reward.reward_rate.amount.into());
-		let new_total_rewards =
-			newly_accumulated_rewards.saturating_add(reward.total_rewards.into());
+		let newly_accumulated_rewards = releasable_periods_surpassed
+			.safe_mul(&reward.reward_rate.amount.into())
```

Comment:
There's no need for fallible arithmetic here (see this comment: https://github.com/ComposableFi/composable/pull/2126#discussion_r1008393598)